src/Pretzel.Logic/Commands/ICommandParametersExtentions.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using Pretzel.Logic.Extensibility;
+
+namespace Pretzel.Logic.Commands
+{
+    public static class ICommandParametersExtentions
+    {
+        public static IEnumerable<ExportFactory<IHaveCommandLineArgs, CommandArgumentsExtentionAttribute>> GetCommandExtentions(this ICommandParametersExtendable commandParametersExtendable)
+        {
+            var attr = commandParametersExtendable.GetType().GetCustomAttributes(typeof(CommandArgumentsAttribute), true).FirstOrDefault();
+
+            if (attr is CommandArgumentsAttribute commandArgumentAttribute && commandParametersExtendable.ArgumentExtenders != null)
+            {
+                foreach (var factory in commandParametersExtendable.ArgumentExtenders)
+                {
+                    if (factory.Metadata.CommandNames.Contains(commandArgumentAttribute.CommandName))
```

Comment:
I am not a fan to rely on strings, isn't it possible to use generics instead?

Reply:
I thought about this one, it would only be possible if we expose all the argument types as interfaces in Pretzel.Logic. When using the export factory with metadata (without creating the instance of the extender) it is impossible to know which argument to extend.

Reply:
This will also solve the problem for plugin authors to get the correct argument instance imported (so no double binding of the command arguments is needed)

Reply:
Is it not possible to use classes directly in order to avoid having to create some more interfaces?

Reply:
I don't like the idea of using the base class for exporting. I refactored out some stuff on this. Let's look if the new design is clearer now. (Need to fix some tests first, let you know if this part is ready for review)

Reply:
Should be done.